internal/common/util/hash.go

```diff
@@ -0,0 +1,502 @@
+// This code is largely a copy from https://github.com/mitchellh/hashstructure/blob/master/hashstructure.go
```

Comment:
Why can't we just reuse this function?  Esp since you are pulling it in for go.mod and go.sum?

Reply:
Reuse which function? 

We had to modify the Hash function to special case resource.Quantity as there is no way to extend it
 - Without it - hashing doesn't work for resource.Quantity 

We plan to ask the library if they can allow extension in some way - so we don't have to copy their code to make a small tweak

Reply:
I see.